sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1484,7 +1476,7 @@ class Analyzer(override val catalogManager: CatalogManager)
           }
         )
       case g: Generate if containsStar(g.generator.children) =>
-        failAnalysis("Invalid usage of '*' in explode/json_tuple/UDTF")
+        throw QueryCompilationErrors.invalidStarUsageError(g.generator.prettyName)
```

Comment:
There is one unit test that matches this error msg: `DataFrameSuite.Star Expansion - ds.explode should fail with a meaningful message if it takes a star`. To avoid changing the error message we can use `invalidStarUsageError("explode/json_tuple/UDTF")` for this one and `invalidStarUsageError(s"expression '${o.prettyName}'")` for the one below.

Reply:
Thank you